mcs/class/System/System.Net.WebSockets/WebSocketHandle.wasm.cs

```diff
@@ -0,0 +1,108 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Runtime.ExceptionServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.WebSockets
+{
+    internal sealed class WebSocketHandle
+    {
+        private WebSocketState _state = WebSocketState.Connecting;
+        private WebAssembly.Net.WebSockets.ClientWebSocket _webSocket;
+
+        public static WebSocketHandle Create() => new WebSocketHandle();
+
+        public static bool IsValid(WebSocketHandle handle) => handle != null;
+
+        public WebSocketCloseStatus? CloseStatus => _webSocket?.CloseStatus;
+
+        public string CloseStatusDescription => _webSocket?.CloseStatusDescription;
+
+        public WebSocketState State => _webSocket?.State ?? _state;
+
+        public string SubProtocol => _webSocket?.SubProtocol;
+
+        public static void CheckPlatformSupport() { /* nop */ }
+
+        public void Dispose()
+        {
+            _state = WebSocketState.Closed;
+            _webSocket?.Dispose();
+        }
+
+        public void Abort()
+        {
+            _webSocket?.Abort();
+        }
+
+        public Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType messageType, bool endOfMessage, CancellationToken cancellationToken) =>
+            _webSocket.SendAsync(buffer, messageType, endOfMessage, cancellationToken);
+
+        public ValueTask SendAsync(ReadOnlyMemory<byte> buffer, WebSocketMessageType messageType, bool endOfMessage, CancellationToken cancellationToken) =>
+            _webSocket.SendAsync(buffer, messageType, endOfMessage, cancellationToken);
+
+        public Task<WebSocketReceiveResult> ReceiveAsync(ArraySegment<byte> buffer, CancellationToken cancellationToken) =>
+            _webSocket.ReceiveAsync(buffer, cancellationToken);
+
+        public ValueTask<ValueWebSocketReceiveResult> ReceiveAsync(Memory<byte> buffer, CancellationToken cancellationToken) =>
+            _webSocket.ReceiveAsync(buffer, cancellationToken);
+
+        public Task CloseAsync(WebSocketCloseStatus closeStatus, string statusDescription, CancellationToken cancellationToken) =>
+            _webSocket.CloseAsync(closeStatus, statusDescription, cancellationToken);
+
+        public Task CloseOutputAsync(WebSocketCloseStatus closeStatus, string statusDescription, CancellationToken cancellationToken) =>
+            _webSocket.CloseOutputAsync(closeStatus, statusDescription, cancellationToken);
+
+        public async Task ConnectAsyncCore(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options)
+        {
+            // TODO #14480 : Not currently implemented, or explicitly ignored:
+            // - ClientWebSocketOptions.UseDefaultCredentials
```

Comment:
We should detect the user providing non-default values for any of these unimplemented options and throw

Reply:
In particular, requesting a proxy and silently performing an unproxied connection can have significant negative consequences

Reply:
You can consider this suggestion optional for now, we just need to do it eventually

Reply:
The `ClientWebSocketOptions` throws not implemented for these.